tools/common/new-release.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2021, Arm Limited
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# invocation example:
+# ./tools/common/new-release.py . 2021 Q1 "02\ :sup:\`nd\` March"
+
+import argparse
+import os
+import re
+
+description = (
+    "Replace dates in ABI files with dates suitable for a new release.")
+
+
+def convert(text, quarter, year, date_of_issue):
```

Comment:
minor: In my experience, "touch all" scripts like this to update copyright years can be disruptive to downstream consumers if there are a lot of otherwise unchanged files. It's also not strictly necessary. That's probably not an issue for this project, though.

Reply:
Yes, technically I would have preferred a central variable (of course), but the decision was made to have a strict separation between the ABI files in their own directory so they could be easier consumed individually. In this case I'd rather stick to not breaking that contract. I agree that your sketched problem is most probably not an issue here.